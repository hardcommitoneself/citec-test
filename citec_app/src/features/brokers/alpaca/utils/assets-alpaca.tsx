// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { AlpacaApi } from "./hooks/api/alpaca-api"

export const assetsAlpaca = async (type: string, tickers: string[]) => {
    
    const getAlpacaAsset = async (type: string, ticker: string) => {
        try {
            const asset = await AlpacaApi({
                type: type,
                url: 'assets/' + ticker,
                method: 'GET',
             })
      
             return asset.data
        } catch (error: any) {
            return error
        }  
    }

    const getAllAssets = async (type: string, tickers: string[]) => {
        
        const assetsPromises: any = []
        tickers.map((ticker) => {
            assetsPromises.push(getAlpacaAsset(type, ticker))
        })

        return await Promise.all(assetsPromises)
    }
    
    return getAllAssets(type, tickers)    
}