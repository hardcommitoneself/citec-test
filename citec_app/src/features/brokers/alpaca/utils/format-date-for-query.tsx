// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
export const formatDateforQuery = (date: Date) => {
    const dformat = [ date.getFullYear(),
        padLeft(date.getMonth()+1),
        padLeft(date.getDate())
        ].join('-');
    
    return dformat
}

function padLeft(n: number){
    return ("00" + n).slice(-2);
  }
 
  
