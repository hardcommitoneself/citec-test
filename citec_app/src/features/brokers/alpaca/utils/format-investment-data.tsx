// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

export const formatInvestmentData = (data: any) => {
    return {'Asset class': data.asset_class,
    'quantity': Number(data.quantity),
    'price': Number(data.price), 
    'Investment': Number(data.investment),  
    'Day gain/loss ($)': Number(data.day_gain_loss[0]),
    'Day gain/loss (%)': Number(data.day_gain_loss[1]),
    'Total gain/loss ($)': Number(data.total_gain_loss[0]),
    'Total gain/loss (%)': Number(data.total_gain_loss[1]),
    }
}
