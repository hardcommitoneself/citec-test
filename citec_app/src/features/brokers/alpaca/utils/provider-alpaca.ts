// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import axios from 'axios'

export const ProviderAlpaca = {
   SetTokenAuth0Alpaca(oauth_code: string) {
      return ProviderAlpaca.setTokenAlpaca(oauth_code)
   },

   async getToken(oauth_code: string) {
      const body = {
         grant_type: 'authorization_code',
         code: oauth_code,
         client_id: process.env.REACT_APP_ALPACA_CLIENT_ID as string,
         client_secret: process.env.REACT_APP_ALPACA_CLIENT_SECRET as string,
         redirect_uri: process.env.REACT_APP_ALPACA_REDIRECT_URI as string,
      }

      const objectBody = Object.keys(body) as Array<keyof typeof body>

      const encodedBody = objectBody
         .map((key) => `${key}=${encodeURIComponent(body[key])}`)
         .join('&')

      let data = {
         code: '',
         value: '',
      }

      await axios({
         headers: {
            'Content-Type': 'application/x-www-form-urlencoded',
         },
         method: 'POST',
         url: 'https://api.alpaca.markets/oauth/token',
         data: encodedBody,
      })
         .then(function (response) {
            data = {
               code: 'sucess',
               value: response.data.access_token,
            }
         })
         .catch(function (error) {
            data = {
               code: 'error',
               value: error,
            }
         })

      return data
   },

   async setTokenAlpaca(code: string) {
      const token = await ProviderAlpaca.getToken(code)
      return token
   },
}
