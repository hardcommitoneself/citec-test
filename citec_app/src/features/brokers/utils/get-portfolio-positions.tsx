// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { fetchDataFromCitecApi } from "features/utils/api/citec-api"
import { store } from "store"

export const getPortfolioPositions = async (tickers: string[]) => {
    
    const outstaing_balance = Array(tickers.length).fill(0)
    const unrealized_capital_gains = Array(tickers.length).fill(0)
    const porfolioOptimized = store.getState().portfolio_optimized

    try {
        const positions = await fetchDataFromCitecApi<any>({
            endpoint: '/ai/portfolio/positions',
            method: 'POST',
            payload: {
                portfolio: 'myportfolio',
                benchmark: porfolioOptimized?.benchmark,
                ticker: tickers,
                outstanding_balance: outstaing_balance,
                unrealized_capital_gains: unrealized_capital_gains,
            },
        })

        return positions.data
    } catch (error: any) {
        return error
    }
}