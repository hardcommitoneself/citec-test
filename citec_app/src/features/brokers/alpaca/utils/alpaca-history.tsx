// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { AlpacaApi } from 'features/brokers/alpaca/utils/hooks/api/alpaca-api'
import { formatDateforQuery } from 'features/brokers/alpaca/utils/format-date-for-query'
import { formatHistoryData } from 'features/brokers/alpaca/utils/format-history-data'

export const AlpacaHistory = (
   typeAccount: string,
   typeHistory: string,
   callback: any,
   callbackError: any
) => {
   const today = new Date()

   const params = {
      timeframe: '1D',
      date_end: formatDateforQuery(today),
      extended_hours: true,
      period: '1D',
      date_start: '',
   }

   switch (typeHistory) {
      case '1D': {
         params.timeframe = '5Min'
         params.period = '1D'
         break
      }
      case '1W': {
         params.timeframe = '1D'
         params.period = '1W'
         break
      }
      case '1M': {
         params.period = '1M'
         break
      }

      case '3M': {
         params.period = '3M'
         break
      }
      case '1Y': {
         params.period = '1A'
         break
      }
      case 'YTD': {
         params.period = '1A'
         params.date_end = ''
         params.date_start = formatDateforQuery(new Date(today.getFullYear(), 0, 1))
         break
      }

      default:
   }

   AlpacaApi({
      type: typeAccount,
      url: 'account/portfolio/history',
      method: 'GET',
      params: params,
   })
      .then(function (response) {
         if (response.data.equity.length == 0) {
            setDataHistory([], 0, 0, callback)
         } else {
            const actualEquity = getActualEquity(response.data.equity)
            const firstEquity = getFirstEquityValid(response.data.equity)

            const equityFinal = getFinalEquityWhitouthZero(
               response.data.equity,
               firstEquity as number
            )

            const finalTimestamp = getFinalTimestampMiliseconds(
               response.data.timestamp,
               typeHistory
            )
            const datos = formatHistoryData(finalTimestamp, equityFinal)

            if (typeHistory == '1D') {
               const params = {
                  timeframe: '1D',
                  date_end: formatDateforQuery(today),
                  extended_hours: true,
                  period: '1W',
               }

               AlpacaApi({
                  type: typeAccount,
                  url: 'account/portfolio/history',
                  method: 'GET',
                  params: params,
               })
                  .then(function (res) {
                     const lastPosition = res.data.equity.length - 2
                     const firstEquity = res.data.equity[lastPosition]
                     setDataHistory(datos, actualEquity, firstEquity, callback)
                  })
                  .catch(function (err) {
                     callbackError(err)
                  })
            } else {
               setDataHistory(datos, actualEquity, firstEquity, callback)
            }
         }
      })
      .catch(function (error) {
         callbackError(error)
      })
}

export function getFinalEquityWhitouthZero(
   arrayEquity: number[],
   value: number
): number[] {
   const equityFinal = arrayEquity.map(function (eq: number) {
      return eq == 0 ? value : eq
   })

   return equityFinal
}

export function getFinalTimestampMiliseconds(
   arrayTimestamp: number[],
   typeH: string
): number[] {
   if (typeH == '1D') return arrayTimestamp

   const dayInSeconds = 60 * 60 * 24 * 1
   const finalTimestamp = arrayTimestamp.map(function (timestmp, index) {
      return index != arrayTimestamp.length - 1 ? timestmp - dayInSeconds : timestmp
   })

   return finalTimestamp
}

export function getActualEquity(arrayEquity: number[]) {
   const filteredEquitys = arrayEquity.filter((equity: number) => equity !== null)
   const last = filteredEquitys.length - 1

   return filteredEquitys[last]
}

export function getFirstEquityValid(arrayEquity: number[]): number {
   const firstNumber = arrayEquity.find((element: number) => {
      return element != 0
   })

   return firstNumber as number
}

function setDataHistory(
   datos: number[][],
   ActualEquity: number,
   firstEquity: number,
   callback: any
) {
   const profitLoss = ActualEquity - (firstEquity as number)
   const profitLossPer =
      ((ActualEquity - (firstEquity as number)) / (firstEquity as number)) * 100

   callback(datos, ActualEquity, profitLoss, profitLossPer)
}
