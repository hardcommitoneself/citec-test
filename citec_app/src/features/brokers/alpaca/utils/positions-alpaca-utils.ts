// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import { MyInvestments } from 'features/brokers/alpaca/ui/components/alpaca-my-invest'
import { numberToFixed } from 'features/utils/to-fixed-number'

export function getStockForValidation(dataPosition: any) {
   const stocks: string[] = []

   if (dataPosition) {
      dataPosition.map((pos: any) => {
         if (pos.asset_class != 'crypto') stocks.push(pos.symbol)
      })
   }

   return stocks
}

export function getTypeStock(stock: string, arrayStocks: any) {
   const searchIndex = arrayStocks.findIndex((stc: any) => stc.ticker == stock)
   return arrayStocks[searchIndex].asset_type
}

export function buildDataInvest(dataPosition: any, DataStock: any) {
   const us_equity_positions: MyInvestments[] = []
   const crypto_positions: MyInvestments[] = []
   const etf_positions: MyInvestments[] = []
   let us_equity_sum = {
      qtys: 0,
      costo: 0,
      invest: 0,
      profit_loss_total: 0,
      profit_los_day: 0,
   }
   let crypto_sum = {
      qtys: 0,
      costo: 0,
      invest: 0,
      profit_loss_total: 0,
      profit_los_day: 0,
   }

   let etf_sum = {
      qtys: 0,
      costo: 0,
      invest: 0,
      profit_loss_total: 0,
      profit_los_day: 0,
   }

   const allowed_types = ['Common Stock', 'REIT']

   if (dataPosition) {
      dataPosition.map((position: any, index: number) => {
         const data = {
            asset_class: position.symbol,
            quantity: position.qty,
            price: position.current_price,
            investment: position.market_value,
            day_gain_loss: [
               position.unrealized_intraday_pl,
               numberToFixed(Number(position.unrealized_intraday_plpc) * 100),
            ],
            total_gain_loss: [
               position.unrealized_pl,
               numberToFixed(Number(position.unrealized_plpc) * 100),
            ],
            url_logo: DataStock[index]?.url_logo,
            subRows: undefined,
         }
   
         if (position.asset_class == 'us_equity') {
            const type = getTypeStock(position.symbol, DataStock)
            if (allowed_types.includes(type)) {
               us_equity_positions.push(data)
               us_equity_sum = {
                  qtys: Number(us_equity_sum.qtys) + Number(position.qty),
                  costo: Number(us_equity_sum.costo) + Number(position.cost_basis),
                  invest: Number(us_equity_sum.invest) + Number(position.market_value),
                  profit_loss_total:
                     Number(us_equity_sum.profit_loss_total) +
                     Number(position.unrealized_pl),
                  profit_los_day:
                     Number(us_equity_sum.profit_los_day) +
                     Number(position.unrealized_intraday_pl),
               }
            } else {
               etf_positions.push(data)
               etf_sum = {
                  qtys: Number(etf_sum.qtys) + Number(position.qty),
                  costo: Number(etf_sum.costo) + Number(position.cost_basis),
                  invest: Number(etf_sum.invest) + Number(position.market_value),
                  profit_loss_total:
                     Number(etf_sum.profit_loss_total) + Number(position.unrealized_pl),
                  profit_los_day:
                     Number(etf_sum.profit_los_day) +
                     Number(position.unrealized_intraday_pl),
               }
            }
         } else {
            crypto_positions.push(data)
            crypto_sum = {
               qtys: Number(crypto_sum.qtys) + Number(position.qty),
               costo: Number(crypto_sum.costo) + Number(position.cost_basis),
               invest: Number(crypto_sum.invest) + Number(position.market_value),
               profit_loss_total:
                  Number(crypto_sum.profit_loss_total) + Number(position.unrealized_pl),
               profit_los_day:
                  Number(crypto_sum.profit_los_day) +
                  Number(position.unrealized_intraday_pl),
            }
         }
      })
   }

   let us_equity = {} as MyInvestments
   if (us_equity_positions.length > 0) {
      us_equity = {
         asset_class: 'US Equities',
         quantity: us_equity_sum.qtys,
         price: 0,
         investment: us_equity_sum.invest,
         day_gain_loss: [
            us_equity_sum.profit_los_day,
            numberToFixed(
               (us_equity_sum.profit_los_day /
                  (us_equity_sum.invest - us_equity_sum.profit_los_day)) *
                  100
            ),
         ],
         total_gain_loss: [
            us_equity_sum.profit_loss_total,
            numberToFixed(
               (us_equity_sum.profit_loss_total / us_equity_sum.costo) * 100
            ),
         ],

         subRows: us_equity_positions,
      }
   }
   let crypto = {} as MyInvestments
   if (crypto_positions.length > 0) {
      crypto = {
         asset_class: 'Crypto',
         quantity: crypto_sum.qtys,
         price: 0,
         investment: crypto_sum.invest,
         day_gain_loss: [
            crypto_sum.profit_los_day,
            numberToFixed(
               (crypto_sum.profit_los_day /
                  (crypto_sum.invest - crypto_sum.profit_los_day)) *
                  100
            ),
         ],
         total_gain_loss: [
            crypto_sum.profit_loss_total,
            numberToFixed((crypto_sum.profit_loss_total / crypto_sum.costo) * 100),
         ],
         subRows: crypto_positions,
      }
   }

   let etf = {} as MyInvestments
   if (etf_positions.length > 0) {
      etf = {
         asset_class: 'ETF',
         quantity: etf_sum.qtys,
         price: 0,
         investment: etf_sum.invest,
         day_gain_loss: [
            etf_sum.profit_los_day,
            numberToFixed(
               (etf_sum.profit_los_day / (etf_sum.invest - etf_sum.profit_los_day)) *
                  100
            ),
         ],
         total_gain_loss: [
            etf_sum.profit_loss_total,
            numberToFixed((etf_sum.profit_loss_total / etf_sum.costo) * 100),
         ],
         subRows: etf_positions,
      }
   }
   const respuesta = []

   if (Object.entries(us_equity).length > 0) respuesta.push(us_equity)
   if (Object.entries(crypto).length > 0) respuesta.push(crypto)
   if (Object.entries(etf).length > 0) respuesta.push(etf)

   return respuesta
}
