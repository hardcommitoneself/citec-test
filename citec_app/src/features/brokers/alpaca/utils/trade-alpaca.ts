// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import { AlpacaApi } from 'features/brokers/alpaca/utils/hooks/api/alpaca-api'
import { OrderAlpaca } from 'features/brokers/alpaca/utils/trades-to-order-alpaca'

export const TradeAlpaca = (type: string, order: OrderAlpaca) => {
   const getTradeAlpaca = async (type: string, order: OrderAlpaca) => {
      try {
         const data = await AlpacaApi({
            type: type,
            url: 'orders',
            method: 'POST',
            payload: JSON.stringify(order),
         })
         
         return data
      } catch (error: any) {
         if (error.response) {
            console.log('error', order)
            console.log(error.response.data)
            console.log(error.response.status)
            console.log(error.response.headers)
         }
      }
   }

   return getTradeAlpaca(type, order)
}
