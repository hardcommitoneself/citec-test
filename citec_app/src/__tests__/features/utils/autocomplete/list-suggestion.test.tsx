// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { fireEvent, render } from '@testing-library/react'
import { ListSuggestions } from 'features/utils/autocomplete'

describe('test at ListSuggestion component', () => {
   const mockTickers = [
      { name: 'Tesla', ticker: 'TSLA' },
      { name: 'Apple', ticker: 'AAPL' },
   ]

   const mockHandleSelectValue = jest.fn()
   const mockHandleOverList = jest.fn()

   beforeEach(() => {
      // eslint-disable-next-line @typescript-eslint/no-empty-function
      Element.prototype.scrollTo = () => {}
   })

   test('should match snapshot', () => {
      const { container } = render(
         <ListSuggestions
            isLoading={false}
            renderSuggestions={true}
            allTickers={mockTickers as never}
            handleSelectValue={mockHandleSelectValue}
            handleOverList={mockHandleOverList}
            suggestionIndex={0}
         />
      )

      expect(container).toMatchSnapshot()
   })

   test('should render No tickers found if allTicker is []', () => {
      const { getByTestId } = render(
         <ListSuggestions
            isLoading={false}
            renderSuggestions={true}
            allTickers={[]}
            handleSelectValue={mockHandleSelectValue}
            handleOverList={mockHandleOverList}
            suggestionIndex={0}
         />
      )

      const textNotFound = getByTestId('not-found')

      expect(textNotFound).toBeDefined()
   })

   test('should render Please type the ticker if allTicker is null', () => {
      const { getByTestId } = render(
         <ListSuggestions
            isLoading={false}
            renderSuggestions={true}
            allTickers={null}
            handleSelectValue={mockHandleSelectValue}
            handleOverList={mockHandleOverList}
            suggestionIndex={0}
         />
      )

      const textTypeTicker = getByTestId('type-ticker')

      expect(textTypeTicker).toBeDefined()
   })

   test('should call mockHandleSelectValue when selec ticker', () => {
      const { getByTestId } = render(
         <ListSuggestions
            isLoading={false}
            renderSuggestions={true}
            allTickers={mockTickers as never}
            handleSelectValue={mockHandleSelectValue}
            handleOverList={mockHandleOverList}
            suggestionIndex={0}
         />
      )

      const buttonTsla = getByTestId('button-TSLA')

      fireEvent.click(buttonTsla)

      expect(mockHandleSelectValue).toBeCalled()
      expect(mockHandleSelectValue).toBeCalledTimes(1)
      expect(buttonTsla).toBeDefined()
   })
   test('should render loading indicator', () => {
      const { getByTestId } = render(
         <ListSuggestions
            isLoading={true}
            renderSuggestions={true}
            allTickers={[]}
            handleSelectValue={mockHandleSelectValue}
            handleOverList={mockHandleOverList}
            suggestionIndex={0}
         />
      )

      const loadingIndicator = getByTestId('loading')

      expect(loadingIndicator).toBeDefined()
   })
})
