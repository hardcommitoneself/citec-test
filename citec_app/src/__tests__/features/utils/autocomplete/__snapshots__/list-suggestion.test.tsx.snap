// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test at ListSuggestion component should match snapshot 1`] = `
<div>
  <div
    class="absolute top-[62px] w-full bg-white custom_cell py-2 max-h-80 rounded-lg shadow-md selector overflow-y-scroll z-20 scroll"
    style="opacity: 1; transform: scale(1, 1); transform-origin: 0 0 0; transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 333ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <div
      class="flex flex-col items-center justify-center"
    >
      <button
        class="transition-all duration-75 hover:bg-[#0BACA9]/30 m-1 p-1 text-left rounded-md cursor-pointer block w-[96%] bg-[#0BACA9]/30"
        data-testid="button-TSLA"
        tabindex="0"
        type="button"
      >
        TSLA
         - 
        Tesla
      </button>
      <button
        class="transition-all duration-75 hover:bg-[#0BACA9]/30 m-1 p-1 text-left rounded-md cursor-pointer block w-[96%] false"
        data-testid="button-AAPL"
        tabindex="0"
        type="button"
      >
        AAPL
         - 
        Apple
      </button>
    </div>
  </div>
</div>
`;
